newsfragments/3313.fixed.md

```diff
@@ -0,0 +1 @@
+Fix case where `PyErr::matches` and `PyErr::is_instance` returned inconsistent results with `PyErr::get_type()`.
```

Comment:
I am bit confused:

I think in your cover letter (PR description), you mean `PyErr::matches` and `PyErr::is_instance` like here?

But also think you want `returned results inconsistent with` like in your cover letter, not `returned inconsistent results with` which might be read as "the results of `PyErr::matches` are different when `PyErr::get_type` is also used"?

Reply:
Yes it's `matches` and `is_instance` which I am claiming had dodgy behaviour before this PR. I'll fixup, good catch.